tools/project_manager/page_compare.php

```diff
@@ -0,0 +1,203 @@
+{
+    echo "<p>";
+    if ($project->archived != 0)
+    {
+        echo _("The project has been archived, so page details are not available.");
+    }
+    elseif ($project->state == PROJ_DELETE)
+    {
+        echo _("The project has been deleted, so page details are not available.");
+    }
+    else
+    {
+        echo _("Page details are not available for this project.");
+    }
+    echo "</p>";
+    exit();
+}
+
+$round_selector = new RoundSelector($page_set);
+$round_selector->render($projectid, $L_round_id, $R_round_id);
+
+// if this is first entry don't do anything else
+if(!$go_compare)
+{
+    exit();
+}
+
+$state_index = array_flip($PROJECT_STATES_IN_ORDER);
+
+if(not_started($L_round_id) || not_started($R_round_id))
+{
+    exit();
+}
+
+$L_round = get_Round_for_round_id($L_round_id);
+$R_round = get_Round_for_round_id($R_round_id);
+$L_text_column_name = $L_round->text_column_name;
+$R_text_column_name = $R_round->text_column_name;
+$L_round_num = $L_round->round_number;
+$R_round_num = $R_round->round_number;
+
+$right_complete = ($state_index[$project->state] >= $state_index["{$R_round_id}.proj_done"]);
+$username = $pguser;
+
+switch($page_set) {
+    case 'right':
+        $condition = "$R_round->user_column_name = '$username'";
+        break;
+    case 'left':
+        $condition = "$L_round->user_column_name = '$username'";
+        break;
+    default: // all
+        $condition = "1";
+        break;
+}
+
+if(!$right_complete)
+{
+    $condition .= " AND state='$R_round->page_save_state'";
+}
+
+$res = mysqli_query(DPDatabase::get_connection(), "
+    SELECT image, $L_text_column_name, $R_text_column_name
+    FROM $projectid
+    WHERE $condition
+    ORDER BY image ASC
+") or die(mysqli_error(DPDatabase::get_connection()));
+
+$num_rows = mysqli_num_rows($res);
+if($num_rows == 0)
+{
+    echo "<p>", _("There are no pages to compare"), "</p>\n";
+    exit();
+}
+else
+{
+    echo "<p>", sprintf(_("Comparing %d pages"), $num_rows), "</p>\n";
+}
+
+// make an array of imagenames of pages with diffs
+$diff_pages = array();
+for($rownum=0; $rownum < $num_rows; $rownum++)
+{
+    $page_res = mysqli_fetch_assoc($res);
+    // also unwrap
+    $L_text = remove_formatting($page_res[$L_text_column_name], true);
+    $R_text = remove_formatting($page_res[$R_text_column_name], true);
+    if(0 != strcmp($L_text, $R_text))
+    {
+        $diff_pages[] = $page_res['image'];
+    }
+}
+
+if(empty($diff_pages))
+{
+    echo "<p>", _("There are no differences"), "</p>\n";
+    exit();
+}
+
+// Draw the results
+echo "<p>", _("Clicking on a link will show the differences in a new window or tab."), "</p>\n";
+echo "<p>";
+foreach($diff_pages as $imagename)
+{
+    echo "<a href='$code_url/tools/project_manager/diff.php?project=$projectid&amp;image=$imagename&amp;L_round_num=$L_round_num&amp;R_round_num=$R_round_num&amp;format=remove' target='_blank'>$imagename</a>\n";
+}
+echo "</p>";
+
+function not_started($round_id)
```

Comment:
This function could use a better name -- particularly since it's doing two things (returning true/false if it hasn't started and also printing something out if it hasn't). Also better to make it a positive-named function instead of a negative. Perhaps `has_project_started_round()`.